src/Date.php

```diff
@@ -276,6 +288,34 @@ public function format($format)
         return parent::format($format);
     }
 
+     /**
+     * Set array in format [ locale => [month => choise] ], to override default translation logic.
+     *
+     * ex: ['ru' => ['May' => 1] ]
+     *
+     * @param  array               $overrides
+     */
+    public static function setForceLocale($overrides) {
+        static::$overrides = $overrides;
+    }
+
+    /**
+     * Check for overrides and if founded pass choice by ref.
+     *
+     * @param  array               $overrides
+     * @param  int                 &$choice
+     * @return  bool               
+     */
+    protected function checkForceLocale($key, &$choice) {
+        if (array_key_exists($this->getLocale(), static::$overrides)) {
+            if (array_key_exists($key, static::$overrides[$this->getLocale()])) {
+                $choice = static::$overrides[$this->getLocale()][$key];
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
This can be a bit tidier:

```php
protected cuntion checkForceLocale($key, &$choice) {
    if (!array_key_exists($this->getLocale(), static::$overrides)) {
        return false;
    }

    if (array_key_exists($key, static::$overrides[$this->getLocale()])) {
        $choice = static::$overrides[$this->getLocale()][$key];
        return true;
    }
}
```